File: src/main/java/com/tom/student/Student.java
package com.tom.student;

public class Student {
    String name;
    String id;
    int english;
    int math;
    static int pass= 60;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public void print(){
        String temp=  name+ "\t"+ english +"\t" +math +"\t"+ getAverage()+
                "\t"+ ((getAverage()>=pass)? "PASS": "FAILED");
        System.out.print(temp);

        int average= getAverage();
        char grading= 'F';
        switch(average/ 10){
            case 10:
            case 9:
                grading= 'A';
                break;
            case 8:
                grading= 'B';
                break;
            case 7:
                grading= 'C';
                break;
            case 6:
                grading= 'D';
                break;
            default:
                grading='F';
        }
        System.out.println("\t"+ grading);
//        if(getAverage()>= 60){
//            System.out.println("\tPASS");
//        }else{
//            System.out.println("\tFAILED");
//        }
    }

    public int getAverage(){
        return (english+math)/2;
    }

    public int highest(){
//        int max= 0;
//        if(english> math){
//            max= english;
//        }else{
//            max= math;
//        }
//        int max= (english> math)? english: math;
//        return max;
        return (english> math)? english: math;
    }
}

File: src/main/kotlin/com/tom/Student.kt
package com.kotlin
import java.util.*

fun main(args: Array<String>) {
//    userInput()
    val stu = Student("Jack", 77, 99)
    stu.print()
    println("highest score: ${stu.highest()}")
    Student.test()
    val gstu= GraduateStudent("Andy",61,65,63)
    gstu.print()
}

private fun userInput() {
    val scanner = Scanner(System.`in`);
//    print("Please enter student's name:");
//    var name= scanner.next();
    var name: String? = null;
    print("Please enter student's english:");
    var english = scanner.nextInt();
    print("Please enter student's math:");
    var math = scanner.nextInt();

    val stu = Student(name, english, math);
    stu.print();
    stu.nameCheck();

}

class GraduateStudent(name: String?,english: Int,math: Int,var thesis: Int): Student(name,english,math){
    companion object {
        var pass= 70
    }
    override fun print() {
        println("$name\t$english\t$math\t$thesis\t${getAverage()}\t${passOrFailed()}\t${grading()}");
    }

    override fun passOrFailed() = if(getAverage()>= pass) "PASS" else "FAILED";
}

open class Student(var name: String?,var english: Int,var math: Int){
    companion object {
        @JvmStatic
        var pass= 60
        fun test(){
            println("testing")
        }
    }
    open fun print(){
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}");
    }
    fun grading()= when(getAverage()){
        in 90..100-> 'A';
        in 80..89-> 'B';
        in 70..79-> 'C';
        in 60..69-> 'D';
        else-> 'F';
    }
    open fun passOrFailed()= if(getAverage()>= pass) "PASS" else "FAILED";
    fun getAverage()= (english+math)/2;
    fun nameCheck()= name?.length;
    fun highest()= if(english> math) english else math;
}